src/arviz_plots/plots/pair_plot.py

```diff
@@ -0,0 +1,495 @@
+    aes_by_visuals["point_estimate"] = aes_by_visuals.get("point_estimate", {})
+    aes_by_visuals["point_estimate_text"] = aes_by_visuals.get("point_estimate_text", {})
+
+    # scatter
+    scatter_kwargs = copy(visuals.get("scatter", {}))
+    if scatter_kwargs is not False:
+        _, scatter_aes, scatter_ignore = filter_aes(
+            plot_matrix, aes_by_visuals, "scatter", sample_dims
+        )
+
+        if "color" not in scatter_aes:
+            colors = plot_bknd.get_default_aes("color", 1, {})
+            scatter_kwargs.setdefault("color", colors[0])
+
+        if "width" not in scatter_aes:
+            scatter_kwargs.setdefault("width", 0)
+
+        if "alpha" not in scatter_aes:
+            scatter_kwargs.setdefault("alpha", 0.5)
+
+        plot_matrix.map_triangle(
+            scatter_couple,
+            "scatter",
+            triangle=triangle,
+            data=distribution,
+            ignore_aes=scatter_ignore,
+            **scatter_kwargs,
+        )
+
+    # marginal
+    remove_axis_bool = copy(visuals.get("remove_axis", False))
+    if marginal is not False:
+        if stats is None:
+            stats = {}
+        else:
+            stats = stats.copy()
+
+        dist_plot_visuals = {}
+        dist_plot_aes_by_visuals = {}
+        dist_plot_stats = {}
+        marginal_dist_kwargs = copy(visuals.get("dist", {}))
+        marginal_ci_kwargs = copy(visuals.get("credible_interval", False))
+        marginal_point_estimate_kwargs = copy(visuals.get("point_estimate", False))
+        marginal_point_estimate_text_kwargs = copy(visuals.get("point_estimate_text", False))
+
+        dist_plot_visuals["dist"] = marginal_dist_kwargs
+        dist_plot_visuals["credible_interval"] = marginal_ci_kwargs
+        dist_plot_visuals["point_estimate"] = marginal_point_estimate_kwargs
+        dist_plot_visuals["point_estimate_text"] = marginal_point_estimate_text_kwargs
+        dist_plot_visuals["title"] = False
+
+        if remove_axis_bool is False:
+            dist_plot_visuals["remove_axis"] = False
+        dist_plot_visuals["rug"] = False
+
+        dist_plot_aes_by_visuals["dist"] = aes_by_visuals.get("dist", {})
+        dist_plot_aes_by_visuals["credible_interval"] = aes_by_visuals.get("credible_interval", {})
+        dist_plot_aes_by_visuals["point_estimate"] = aes_by_visuals.get("point_estimate", {})
+        dist_plot_aes_by_visuals["point_estimate_text"] = aes_by_visuals.get(
+            "point_estimate_text", {}
+        )
+
+        dist_plot_stats["dist"] = stats.get("dist", {})
+        dist_plot_stats["credible_interval"] = stats.get("credible_interval", {})
+        dist_plot_stats["point_estimate"] = stats.get("point_estimate", {})
+
+        plot_matrix = plot_dist(
+            dt,
+            var_names,
+            filter_vars,
+            group,
+            coords,
+            sample_dims,
+            marginal_kind,
```

Comment:
```suggestion
            distribution,
            sample_dims=sample_dims,
            kind=marginal_kind,
```

We have already done all the filtering/subsetting to get `distribution` and `plot_dist` doesn't need data from any other group, so we can pass distribution to avoid processing the same thing twice.